### DISCUSSION TITLE
How can I get the ausgab dosimetry readings to output in a csv file instead of just in the terminal?

### DISCUSSION THREAD

**Discussion by Sasa-fras** [nrc-cnrc/EGSnrc#1104] [Questions]
The main data I need to work with is the energy deposition data from the ausgab dosimetry readings. Is there any way to get an automatically made csv/txt file of the results instead of having to manually clean the data after either copy and pasting from terminal or piping the output into a text file with "tee"? I know that there is other data collected automatically into the .egsdat file so I assume its possible to do the same with the ausgab data. (side note, what is even in the .egsdat file? its a complete mess.)

**Comment by rtownson** [comment]
When you run on a single core it may not produce an `.egslog` file, because it assumes you are prototyping. Piping the output to a text file is run. Run in parallel and it will automatically generate a combined `.egslog` file that contains the output. Use the `egs-parallel` script with a single core if you want, and it will also create a `.egslog` file for the single job.

For n=4 cores to run in parallel on the local machine:
```
egs-parallel --batch cpu -n4 -v --force -c 'egs_app -i slab'
```

Don't use the .egsdat file. It is intended for use by the code internally - it stores intermediate data (the state of the simulation) batch-by-batch, so that if the job crashes then you're able to restart from the last batch.

**Reply by Sasa-fras** [reply]
So what kind of stuff would be in a .egslog file? I really just want the output of the ausgab dosimetry in a csv file or a txt file where the data is only separated by one space instead of many for typesetting purposes. If the log file is just going to do the same as piping the terminal output into a new txt then I would still have to do manual cleaning of the entire file before it can be conveniently used by a python script for data processing/visualization.

**Reply by rtownson** [reply]
Unfortunately there isn't a csv output built into the dose scoring ausgab object. This would be a nice feature - if you are able to do some coding and add this feature, let me know. Otherwise, I will add it to my (long) to-do list.

**Reply by Sasa-fras** [reply]
I could look into adding the feature, I have no clue how though. Where would I be coding to add that? like in what file would I add the code to make that happen?

**Reply by rtownson** [reply]
In `HEN_HOUSE/egs++/ausgab_objects/egs_dose_scoring`, the `.h` and `.cpp` files. See `reportResults()`. Ideally you would name it automatically like .egslst files etc, but for yourself, you could just hardcode a filename..

**Comment by ftessier** [comment]
It is also possible to generate an `.egslog` file for single-cpu runs by adding the `-b` flag (batch run) to the EGSnrc command.

**Comment by ftessier** [comment]
For converting to `csv` format, you can simply copy the dose scoring text output to, for example, a Google Sheet document and select the option to `Split text to columns` upon pasting. In the terminal, you can use the `sed` program to perform the conversion:

```
sed -r 's/  +/,/g' data.txt > data.csv
```

(that is two space characters before the `+` character). To understand and tweak the regular expression pattern for dose scoring object output, check out this example: https://regex101.com/r/JPKh59/2
